tools/mo/openvino/tools/mo/moc_frontend/extractor.py

```diff
@@ -19,46 +19,95 @@ def decode_name_with_port(input_model: InputModel, node_name: str, framework="")
+        match_pre = re.search(regexp_pre, node_name)
+        # check if there is a match
+        if match_pre:
+            # split node_name into port and name
+            port, name = match_pre.group(1), match_pre.group(2)
+            # check if there is an operation with name
+            node = input_model.get_place_by_operation_name(name)
+            # if there is no operation with provided name and we
+            # did not find tensor before we check if there is a
+            # tensor with name
+            if not node and search_tensor:
+                tensor = input_model.get_place_by_tensor_name(name)
+                if tensor:
+                    found_node_names.append('Tensor:' + tensor.get_names()[0])
+                    found_nodes.append(tensor)
+                    search_tensor = False
+            if node:
+                # if there is an operation with given name, we
+                # add input port
+                node_pre = node.get_input_port(input_port_index=int(port))
+                if node_pre:
+                    found_node_names.append(name)
+                    found_nodes.append(node_pre)
+                # if we are still looking for the tensor
+                # we add one with given port
+                if search_tensor:
+                    tensor = node.get_source_tensor(input_port_index=int(port))
+                    if tensor:
+                        found_node_names.append('Tensor:' + tensor.get_names()[0])
+                        found_nodes.append(tensor)
+                        search_tensor = False
+        # we check for name:port combination
+        regexp_post = r'(.+):(\d+)'
+        match_post = re.search(regexp_post, node_name)
+        if match_post:
+            # split node_name into name and port
+            name, port = match_post.group(1), match_post.group(2)
+            # check if there is an operation with name
+            node = input_model.get_place_by_operation_name(name)
+            if not node and search_tensor:
+                tensor = input_model.get_place_by_tensor_name(name)
+                if tensor:
+                    # node = tensor.get_producing_operation()
+                    found_node_names.append('Tensor:' + tensor.get_names()[0])
+                    found_nodes.append(tensor)
+                    search_tensor = False
+            if node:
+                node_post = node.get_output_port(output_port_index=int(port))
+                if node_post:
+                    found_node_names.append(name)
+                    found_nodes.append(node_post)
+                # if tensor was no found yet
+                if search_tensor:
+                    tensor = node.get_target_tensor(output_port_index=int(port))
+                    if tensor:
+                        found_node_names.append('Tensor:' + tensor.get_names()[0])
+                        found_nodes.append(tensor)
+                        search_tensor = False
```

Comment:
I think there are some duplicating code for pre match and post match cases. We can generalize it

Reply:
ok, I will try to generalize it better